core/db.py

```
@@ -144,4 +271,23 @@ def assert_called_once(self) -> None:
     # Use the raw SQLAlchemy engine to avoid any potential wrapper interference
+    if _ASYNC_ENGINE is not None and _RAW_ENGINE is None:
+        raise RuntimeError(
+            "init_db() cannot be used when only an async engine is configured. Use `await init_db_async()` instead."
+        )
     metadata.create_all(bind=_RAW_ENGINE)
+
+
+async def init_db_async() -> None:
+    """Async variant of :func:`init_db` for async engines."""
+
+    import core.models  # pylint: disable=unused-import
```

Flake8 (7.3.0)

[error] 'core.models' imported but unused

(F401)